inference-engine/src/transformations/src/transformations/common_optimizations/fq_reshape_fusion.cpp

```diff
@@ -0,0 +1,73 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/fq_reshape_fusion.hpp"
+
+#include <memory>
+#include <vector>
+
+#include <ngraph/opsets/opset4.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/rt_info.hpp>
+#include <numeric>
+
+ngraph::pass::FakeQuantizeReshapeFusion::FakeQuantizeReshapeFusion() {
+    const auto fq_node_p = ngraph::pattern::wrap_type<opset4::FakeQuantize>(
+            {ngraph::pattern::wrap_type<opset4::Constant>(), // for weights only
+             ngraph::pattern::any_input(),
+             ngraph::pattern::any_input(),
+             ngraph::pattern::any_input(),
+             ngraph::pattern::any_input()},
+            pattern::consumers_count(1));
+    const auto reshape_node_p = ngraph::pattern::wrap_type<opset4::Reshape>(
+            {fq_node_p, ngraph::pattern::any_input()}, pattern::consumers_count(1));
+
+    ngraph::matcher_pass_callback callback = [=](pattern::Matcher &m) {
+        const auto &pattern_map = m.get_pattern_value_map();
+        const auto fq_node = pattern_map.at(fq_node_p).get_node_shared_ptr();
+        if (fq_node->is_dynamic())
+            return false;
+        const auto &reshape_node = pattern_map.at(reshape_node_p).get_node_shared_ptr();
+        const auto &reshape_pattern_node = reshape_node->get_input_node_shared_ptr(1);
```

Comment:
You use reshape_pattern_node only as input for new node. So its better to use Output<Node> instead of Node pointer.
```suggestion
        const auto &reshape_pattern_node = reshape_node->input_value(1);
```